### PR TITLE
Fix argmax bug for batchsize > 1 in arn_generic.py

### PR DESCRIPTION
**Pull request by akrentz6**
`xlns.argmax` doesn't pass the `axis` parameter through to `xlns.xlnsapplynpfunc` which causes it to default to axis=0. In particular, this leads to bugs with `minibatch_size` > 1 in `arn_generic.py`. This is fixed by passing `**kwargs` to `xlns.xlnsapplynpfunc` and then passing those through to the corresponding functions. 

In `arn_generic.py`, I changed standard multiplication symbols to matrix multiplication in the backpropagation as in the original code from Arnab Sanyal to allow for `minibatch_size` > 1.
 
I also added a `--val_split` argument (also defaulting to 50) in addition to the `--split` argument which allows you to choose the number of images used for validation and these validation images are taken from the previously unused test data from `mnist.npz`. This allows for up to the entire 60,000 images to be used during training (and up to 10,000 for validation) but doesn't force you to have the same number of training images as validation images (as was the case before).

Finally, there was inconsistent indent sizes so I adjusted those.

### CODE REVIEW

**Comment by markgarnold** [comment]
Thanks for spotting this issue. @pradeeban and I will review.

File: examples/arn_generic.py [hunk 1]
```diff
@@ -206,36 +207,36 @@ def main(main_params):
...
-				  lnsx = (np.array(xl.xlnscopy(np.array(x,dtype=np.float64))))
+					lnsx = (np.array(xl.xlnscopy(np.array(x,dtype=np.float64))))
 				if main_params['type'] == 'xlnsud':
-				  lnsx = (np.array(xl.xlnscopy(np.array(x,dtype=np.float64),xl.xlnsud)))
+					lnsx = (np.array(xl.xlnscopy(np.array(x,dtype=np.float64),xl.xlnsud)))
 				if main_params['type'] == 'xlnsv':
-				  lnsx = (np.array(xl.xlnscopy(np.array(x,dtype=np.float64),xl.xlnsv)))
+					lnsx = (np.array(xl.xlnscopy(np.array(x,dtype=np.float64),xl.xlnsv)))
 				if main_params['type'] == 'xlnsb':
-				  lnsx = (np.array(xl.xlnscopy(np.array(x,dtype=np.float64),xl.xlnsv,2**2**-xl.xlnsF)))
```
Comment (pradeeban) [inline examples/arn_generic.py:226]: Lots of line changes like this? Can you avoid these and re-create a pull request?

**Comment by pradeeban** [comment]
The major criticism I have for this pull request is,
the majority of code changes seem to be linespace -> tab and removal of End of Line character.

@akrentz6 can you please recreate the pull request with the minimal line change necessary? Then, the pull request will be neat and minimal - and one that will be easy to review and merge.

This is for two reasons:
1. Makes it easy to review now before merge. If something goes wrong later, we know through git history / git blame, which line was authored/changed last by an author. With these unnecessary line changes, issues unrelated to this pull request will come to you!
2. Coding best practices. A pull request should do what it claims to do in its definition. Nothing more, nothing less. If you have a reason to make these style changes, that should be a separate pull request. Not one that is piggybacking on the "Fix argmax bug."

File: examples/arn_generic.py [hunk 2]
```diff
@@ -56,75 +56,76 @@ def main(main_params):
...
 		x_train = x_train[:split]
 
 
 		if os.path.isfile("./weightin.npz"):
-		    print("using ./weightin.npz")
-		    randfile = np.load("./weightin.npz","r")
-		    W1 = randfile["W1"] 
-		    W2 = randfile["W2"]
-		    randfile.close()
+			print("using ./weightin.npz")
```
Comment (pradeeban) [inline examples/arn_generic.py:68]: I am referring to line changes such as this one.

File: examples/arn_generic.py [hunk 3]
```diff
@@ -206,36 +207,36 @@ def main(main_params):
...
 				lnsa1 = lnss1 * lnsmask
 				lnss2 = xl.hstack((lnsones, lnsa1)) @ lnsW2
 				lnscorrect_count += np.sum(np.argmax(y, axis=1) == xl.argmax(lnss2, axis=1))
-			lnsaccuracy = lnscorrect_count / split
+			lnsaccuracy = lnscorrect_count / val_split
 			print("Val-set accuracy at epoch %d: %f" % ((1 + epoch), lnsaccuracy))
 			performance['lnsacc_val'][epoch] = 100 * lnsaccuracy
 		print("elasped time="+str(time.process_time()-start_time))
-		fig = plt.figure(figsize = (16, 9)) 
+		fig = plt.figure(figsize = (16, 9))
```
Comment (pradeeban) [inline examples/arn_generic.py:239]: and this... removal of end of line character.
